core/src/test/kotlin/WatchTest.kt

```diff
@@ -1,31 +1,78 @@
     @JvmStatic
-    fun main(args: Array<String>) {
+    fun main(args: Array<String>) = runBlocking {
         val path = File("pages").toPath()
+        val files = File("build").resolve("test_data").run {
+            deleteRecursively()
+            mkdirs()
+            listOf(resolve("data.txt"), resolve("other.txt"))
+        }
+
+        val jobs = files.map { startWatcher(it.parentFile.toPath()) }
+
+        files.forEach {
+            it.parentFile.mkdirs()
+            it.writeText("${it.name} created")
+        }
+
+        delay(1000)
 
-        val watcher = path.watch()
-        logger.info("Press ctrl+c to exit")
+        files.forEach {
+            it.parentFile.mkdirs()
+            it.writeText("${it.name.toUpperCase()} modified")
+        }
+        files.forEach {
+            it.parentFile.mkdirs()
+            it.writeText("${it.name.toUpperCase()} modified again")
+        }
+        delay(1000)
+        (0..50).forEach {
+            delay(200)
+        }
+    }
 
-        while (true) {
-            //The watcher blocks until an event is available
-            val key = watcher.take()
+    private fun startWatcher(target: Path): Job {
+        logger.debug("starting watcher for $target")
+//        watchJob?.cancel()
+        val watchJob = watchActor(target) {
+            var timeout: Job? = null
+            for (watchEvent in channel) {
+                val path = watchEvent.context()
+                val file = target.toFile().resolve(path.toFile())
+                val event = watchEvent.kind()
+                timeout?.cancel()
+                timeout = launch {
+                    logger.debug("event: $path, ${event.name()}")
+                    delay(1000)
 
-            //Now go through each event on the folder
-            key.pollEvents().forEach { it ->
-                //Print output according to the event
-                when (it.kind().name()) {
-                    "ENTRY_CREATE" -> logger.info("${it.context()} was created")
-                    "ENTRY_MODIFY" -> logger.info("${it.context()} was modified")
-                    "OVERFLOW" -> logger.info("${it.context()} overflow")
-                    "ENTRY_DELETE" -> logger.info("${it.context()} was deleted")
+                    when (watchEvent.kind().name()) {
+                        "ENTRY_CREATE" -> {
+                            logger.debug("$path was created")
+                            logger.info("content: '${file.readText()}'")
+                        }
+                        "ENTRY_MODIFY" -> {
+                            logger.debug("$path was modified")
+                            logger.info("content: '${file.readText()}'")
+                        }
+                        "ENTRY_DELETE" -> {
+                            logger.debug("$path was deleted")
+                        }
+                        "OVERFLOW" -> logger.debug("${watchEvent.context()} overflow")
+                    }
```

Comment:
oh hmmm don't we have jobWatcher in the original code already doing this or nah?

Reply:
this is Test code, not actually compiled in, i just needed to verify hpw the PathWatcher acts